### PR TITLE
Add srfi recipe

### PR DESCRIPTION
**Pull request by lassik**
### Brief summary of what the package does

Provides quick access to Scheme Requests for Implementation (SRFI)
documents from within Emacs:

* `M-x srfi-list` brings up a `*SRFI*` buffer listing all SRFIs.

* `M-x srfi` does the same, but allows live-narrowing the list by
  typing numbers or words into the minibuffer.

* The `*SRFI*` buffer provides one-key commands to visit the SRFI
  document, its discussion archive (mailing list archive), and it
  version control repository. These commands open the right web
  pages in your web browser using `browse-url`.

### Direct link to the package repository

https://github.com/srfi-explorations/emacs-srfi

### Your association with the package

Wrote it.

### Relevant communications with the upstream package maintainer

None needed

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

### CODE REVIEW

**Comment by riscy** [comment]
Quick first pass --

### srfi-data.el ###

byte-compile-file (using Emacs 26.3):
- No issues!

checkdoc (using version 0.6.1):
- No issues!

(Skipping package-lint on this file)


### srfi.el ###

byte-compile-file (using Emacs 26.3):
- No issues!

checkdoc (using version 0.6.1):
- No issues!

package-lint-current-buffer (using version 20200226.2001):
- No issues!

Experimental static checks/suggestions:
- srfi.el#L116: This define-key could overwrite a user's keybindings.  Try: `(defvar my-map (let ((km (make-sparse-keymap))) (define-key ...) km))`
- srfi.el#L117: This define-key could overwrite a user's keybindings.  Try: `(defvar my-map (let ((km (make-sparse-keymap))) (define-key ...) km))`
- srfi.el#L118: This define-key could overwrite a user's keybindings.  Try: `(defvar my-map (let ((km (make-sparse-keymap))) (define-key ...) km))`
- srfi.el#L119: This define-key could overwrite a user's keybindings.  Try: `(defvar my-map (let ((km (make-sparse-keymap))) (define-key ...) km))`
- srfi.el#L120: This define-key could overwrite a user's keybindings.  Try: `(defvar my-map (let ((km (make-sparse-keymap))) (define-key ...) km))`
- Prefer the default recipe, especially for small packages.  I think you can get away with it in this case?

**Comment by lassik** [comment]
> This define-key could overwrite a user's keybindings.

Fixed. Thanks!

**Comment by lassik** [comment]
> Prefer the default recipe, especially for small packages. I think you can get away with it in this case?

srfi-data-convert.el is only useful for package maintainers who need to run it in order to generate srfi-data.el based on the upstream data file srfi-data.scm. (srfi-data-convert.el translates Scheme syntax to Emacs Lisp syntax and throws away information that srfi.el does not need). That's why I specified the MELPA recipe to exclude srfi-data-convert.el

File: recipes/srfi [hunk 1]
```diff
@@ -0,0 +1,2 @@
+(srfi :fetcher github :repo "srfi-explorations/emacs-srfi"
+      :files ("srfi.el" "srfi-data.el"))
```
Comment (conao3) [inline recipes/srfi]: `:files (:default (:exclude "srfi-data-convert.el"))` is better.
Manage with whitelist, you need to modify recipe if you split package file.
Manage with blacklist is you don't need to modify the recipe.

**Comment by riscy** [comment]
Good point re: `:exclude` @conao3.  Another option would be to put that special file inside its own directory, which MELPA would automatically ignore, but that's not a blocker.

I think to be evenhanded in my insistence for license boilerplate, we should consider srfi-data.el to be code, and as such it should have some boilerplate or a SPDX identifier.

Beyond that I think this is good to go, so ping me when ready. :)

**Comment by lassik** [comment]
> Good point re: `:exclude` @conao3. Another option would be to put that special file inside its own directory, which MELPA would automatically ignore, but that's not a blocker.

OK, I moved `srfi-data-convert.el` (and the trivial shell script wrappers assisting it) into an `admin` subdirectory so they are excluded by the default recipe. Force-pushed this PR to update the recipe.

> I think to be evenhanded in my insistence for license boilerplate, we should consider srfi-data.el to be code, and as such it should have some boilerplate or a SPDX identifier.

OK, the generator now outputs `Copyright` and `SPDX-License-Identifier` lines into `srfi-data.el`.

> Beyond that I think this is good to go, so ping me when ready. :)

Thanks a lot!

**Comment by lassik** [comment]
@riscy Oh, and here's the ping :)

**Comment by riscy** [comment]
Thanks!

**Pull request merged**